PINCache/PINDiskCache.m

```diff
@@ -1107,10 +1107,16 @@ - (void)unlock
 @end
 
 @implementation PINBackgroundTask
+
++ (BOOL)isAppExtension
+{
+    return [[[NSBundle mainBundle] executablePath] containsString:@".appex/"];
```

Comment:
Yep this will be dispatched_once. I will look into how we could integrate a unit test for it.